cpp/include/functions.h

```diff
@@ -15,65 +15,189 @@
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in] *edge_data (optional)  This pointer can be nullptr. If not, this gdf_column of size E (number of edges) contains the weiht for each edge.
  *                                   The type expected to be floating point.
- * 
+ *
  * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added edge list (edge_data is optional).
- * 
- * @Returns                          GDF_SUCCESS upon successful completion. 
+ *
+ * @Returns                          GDF_SUCCESS upon successful completion.
  */
 /* ----------------------------------------------------------------------------*/
-gdf_error gdf_edge_list_view(gdf_graph *graph, 
+gdf_error gdf_edge_list_view(gdf_graph *graph,
                              const gdf_column *source_indices,
                              const gdf_column *destination_indices,
                              const gdf_column *edge_data);
 
-gdf_error gdf_adj_list_view (gdf_graph *graph, 
+/**
+ * @Synopsis   Wrap existing gdf columns representing adjacency lists in a gdf_graph.
+ *             cuGRAPH does not own the memory used to represent this graph. This function does not allocate memory.
+ * @Param[in] *offsets               This gdf_column of size V+1 (V is number of vertices) contains the offset of adjacency lists of every vertex.
+ *                                   Offsets must be in the range [0, E] (number of edges).
+ * @Param[in] *indices               This gdf_column of size E contains the index of the destination for each edge.
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in] *edge_data (optional)  This pointer can be nullptr. If not, this gdf_column of size E (number of edges) contains the weiht for each edge.
+ *                                   The type expected to be floating point.
+ *
+ * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added adjacency list (edge_data is optional).
+ *
+ * @Returns                          GDF_SUCCESS upon successful completion.
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_adj_list_view (gdf_graph *graph,
                              const gdf_column *offsets,
                              const gdf_column *indices,
                              const gdf_column *edge_data);
 
+/**
+ * @Synopsis   Create the adjacency lists of a gdf_graph from its edge list.
+ *             cuGRAPH allocates and owns the memory required for storing the created adjacency list.
+ *             The call expects a valid structure pointed by graph->edgeList. Upon successful completion of the call graph->adjList will point to a gdf_adj_list structure containing the generated adjacency list.
+ *
+ * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added adjacency list.
```

Comment:
`graph` is a `param[in, out]`. I would recommend adding the paragraph just above it to explain the [in] part.